### PR TITLE
fix: supported chains endpoint

### LINKED ISSUE #566: API to pull list of supported chains
[Slack conversation](https://walletconnect.slack.com/archives/C04SYT4LU2W/p1709658823506369?thread_ts=1709319338.504139&cid=C04SYT4LU2W)

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Adds a basic endpoint to return a list of supported CAIP-2 chain IDs.

Resolves #566 

## How Has This Been Tested?

New integration test

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/providers/mod.rs [hunk 1]
```diff
@@ -159,6 +161,7 @@ impl ProviderRepository {
         ));
 
         Self {
+            supported_chains: HashSet::new(),
```
Comment (nopestack) [inline src/providers/mod.rs:164]: Consider simply calling `..Default::default` to avoid setting all of these empty HashSets/Maps

**Review by nopestack — APPROVED** [review]
Lgtm

**Review by geekbrother — CHANGES_REQUESTED** [review]
Requesting changes, because I see an issue here:
* Why only ws supported chains were added? In this case, the return should reflect this somehow, e.g. move this list into the `ws` entity of the response.

The test:
`curl -v 'http://localhost:3000/v1/supported-chains?projectId=xxx' | jq .`, the response:
```
[
  "eip155:420",
  "eip155:1",
  "eip155:5",
  "eip155:42161",
  "eip155:10",
  "eip155:11155111",
  "eip155:421613",
  "eip155:7777777"
]
```
There are only ws supported chains.

I think this endpoint should reflect the `SUPPORTED_CHAINS.md`.

**Comment by chris13524** [comment]
@geekbrother oops! Yeah meant to do HTTP RPC not WS supported chains. Good catch.

I suppose I'll put the list of HTTP RPC supported chains in an `http` entity of the response to allow room to support `ws` in the future? WS is deprecated and will probably be removed. Maybe I'll just put both `http` and `ws` in there, might as well and easy enough and models current structure of everything.

File: src/providers/mod.rs (hunk 1 above)
Comment (chris13524) [reply src/providers/mod.rs:164]: `Self` doesn't implement `Default`

**Review by geekbrother — APPROVED** [review]
Nice!

**Pull request merged**